src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -147,13 +149,35 @@ final case class PurposeApiServiceImpl(
     val operationLabel = s"Creating Purpose Version for Purpose $purposeId"
     logger.info(operationLabel)
 
+    def publish(
+      organizationId: UUID,
+      purpose: PersistentPurpose,
+      version: PersistentPurposeVersion
+    ): Future[PurposeManagementDependency.PurposeVersion] = for {
+      eService  <- catalogManagementService.getEServiceById(purpose.eserviceId)
+      ownership <- Ownership
+        .getOrganizationRole(organizationId, eService.producerId, purpose.consumerId)
+        .toFuture
+      published <- purposeVersionActivation.activateOrWaitForApproval(
+        eService,
+        purpose,
+        version,
+        organizationId,
+        ownership
+      )
+    } yield published
+
     val result: Future[PurposeVersion] = for {
       organizationId <- getOrganizationIdFutureUUID(contexts)
       purposeUUID    <- purposeId.toFutureUUID
       purpose        <- purposeManagementService.getPurposeById(purposeUUID)
       _              <- assertOrganizationIsAConsumer(organizationId, purpose.consumerId)
       version        <- purposeManagementService.createPurposeVersion(purposeUUID, seed.toManagement)
-    } yield version.toApi
+      published      <-
+        if (purpose.versions.exists(v => v.state == Active || v.state == Suspended))
+          publish(organizationId, purpose, version.toPersistent)
+        else Future.successful(version)
```

Comment:
if I got it correctly, this is to ensure that the endpoint is called to create a version that it is not the first of the purpose.
It's a nice touch but I think we can simplify it leveraging on the purpose lifecycle. 
When a new purpose is created, the first version is created at the same time, so when the `createPurposeVersion` is invoked, it will be guaranteed to be at least the second version
```suggestion
      published      <- publish(organizationId, purpose, version.toPersistent)
```